rotkehlchen/chain/gnosis/modules/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,104 @@
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.constants import XDAI_BRIDGE_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails, EventCategory
+from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer, bridge_prepare_data
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.chain.gnosis.constants import CPT_XDAI_BRIDGE
+from rotkehlchen.constants.assets import A_XDAI
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, DecoderEventMappingType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+BRIDGE_ADDRESS = string_to_evm_address('0x7301CFA0e1756B71869E93d4e4Dca5c7d0eb0AA6')
+BRIDGE_TOKEN = b'\x12vP\xbc\xfb\x0b\xa0\x17@\x1a\xbeI1E:@Q@\xa8\xfd6\xfe\xceg\xba\xe2\xdb\x17M?\xddc'  # noqa: E501
+
+
+class XdaiBridgeDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.xdai = A_XDAI.resolve_to_crypto_asset()
+
+    def _decode_bridge_tokens(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a bridging event for xdai. Either a deposit or a withdrawal."""
+        if context.tx_log.topics[0] != BRIDGE_TOKEN:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = context.transaction.from_address
+        to_address = BRIDGE_ADDRESS
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.xdai,
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_events=(BRIDGE_TOKEN,),
+            main_chain=ChainID.GNOSIS,
+            l2_chain=ChainID.ETHEREUM,
```

Comment:
@LefterisJP I found this name confusing. Maybe could we rename this to `target_chain` or anything better than `l2`??

Reply:
In that case I would use source/target. And not main/target. Does that make sense?